windows/src/desktop/kmshell/main/OnlineUpdateCheck.pas

```diff
@@ -98,23 +102,141 @@   TOnlineUpdateCheck = class
+      progress bar. This procedure is similar to DownloadUpdates, but it runs in
+      the background.
+
+      @params  SavePath  The path where the downloaded files will be saved.
+               Result    A Boolean value indicating the overall result of the
+               download process.
+    }
+    procedure DoDownloadUpdatesBackground(SavePath: string; var Result: Boolean);
+    {
+      Performs an online update check, including package retrieval and version
+      query.
+
+      This function checks if a week has passed since the last update check. It
+      utilizes the kmcom API to retrieve the current packages. The function then
+      performs an HTTP request to query the remote versions of these packages.
+      The resulting information is stored in the FParams variable. Additionally,
+      the function handles the main Keyman install package.
+
+      @returns  A TOnlineUpdateCheckResult indicating the result of the update
+      check.
+    }
     function DoRun: TOnlineUpdateCheckResult;
+    {
+      Displays the Update form, which allows selection of updates to be manual
+      downloaded.
+
+      This procedure shows the Update form, which contains controls for
+      downloading and performing updates. If elevation of permissions is
+      required for certain operations, it will be handled internally. If the
+      user does not have administrative permissions, a dialog will be raised to
+      inform them to use an account with administration privileges.
+    }
     procedure ShowUpdateForm;
     procedure ShutDown;
+    { Works out the status of the downloaded }
     procedure DownloadUpdatesHTTPStatus(Sender: THTTPUploader;
       const Message: string; Position, Total: Int64); // I2855
-    procedure DoInstallKeyman;
+    {
+      Installs the packages using the kmcom API and deletes the package from the
+      specified SavePath. This procedure utilizes the kmcom API to install the
+      packages.
+    }
+    procedure DoInstallKeyman; overload;
+    function DoInstallKeyman(SavePath: string) : Boolean; overload;
+    {
+      Installs the Keyman file using either msiexec.exe or the setup launched in
+      a separate shell.
+
+      @params  Package  The package to be installed.
+
+      @returns True  if the installation is successful, False otherwise.
+    }
     function DoInstallPackage(Package: TOnlineUpdateCheckParamsPackage): Boolean;
+    {
+      SavePackageUpgradesToDownloadTempPath saves any new package IDs to a
+      single file in the download tempPath. This procedure saves the IDs of any
+      new packages to a file named "upgrade_packages.inf" in the download
+      tempPath.
+    }
     procedure SavePackageUpgradesToDownloadTempPath;
+
+
+   {
+      // RC initially take code from DoRun
+      Performs an online update check, including package retrieval and version
+      query.
+
+      This function checks if a week has passed since the last update check. It
```

Comment:
I think we could update to daily checks? Will help for quick turnaround on bugs.